### PR TITLE
Correctly restore point position after running rustfmt

### PR DESCRIPTION
**Pull request by Wilfred**
The previous approach simply moved to the same character offset. This is
unlikely to preserve the position of point, as rustfmt often changes
whitespace, changing the number of characters before point.

Instead, we go back to the line number and column number we were on
before. Provided that rustfmt has not radically changed the number of
lines, this will typically put point back to its previous position, or
at least close.

Improves, but doesn't completely solve, issue #162.

### CODE REVIEW

File: rust-mode.el [hunk 1]
```diff
@@ -1365,9 +1371,10 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
   ;; to use `font-lock-ensure', which doesn't exist in Emacs 24 and earlier.
   ;; If it's not available, fall back to calling `font-lock-fontify-region'
   ;; on the whole buffer.
-  (if (fboundp 'font-lock-ensure)
-      (font-lock-ensure)
-    (font-lock-fontify-region (point-min) (point-max))))
+  (save-excursion
+    (if (fboundp 'font-lock-ensure)
+        (font-lock-ensure)
+      (font-lock-fontify-region (point-min) (point-max)))))
```
Comment (nikomatsakis) [inline rust-mode.el]: how is this change related?

Comment (Wilfred) [inline rust-mode.el]: `rust-format-buffer` calls this function after repositioning point. If this function moves point, then our repositioning of point doesn't do us any good.

Does that make sense? This is a separate commit, and I'm happy to amend the commit message if I can make it clearer.

**Comment by Wilfred** [comment]
@nikomatsakis ping, are you happy with this? I think it's a worthwhile improvement. Are there any changes you'd like me to make?

**Comment by nikomatsakis** [comment]
Ah, sorry, I agree!
